beets/library.py

```diff
@@ -1502,13 +1504,20 @@ def tmpl_aunique(self, keys=None, disam=None):
 
         else:
             # No disambiguator distinguished all fields.
-            res = u' {0}'.format(album.id)
+            res = u' {1}{0}{2}'.format(album.id, bracket[0] if bracket else
+                                       u'', bracket[1] if bracket else u'')
             self.lib._memotable[memokey] = res
             return res
 
         # Flatten disambiguation value into a string.
         disam_value = album.formatted(True).get(disambiguator)
-        res = u' [{0}]'.format(disam_value)
+        res = u' {1}{0}{2}'.format(disam_value, bracket[0] if bracket else u'',
+                                   bracket[1] if bracket else u'')
+
+        # Remove space and/or brackets if disambiguation value is empty
+        if bracket and res == u' ' + ''.join(bracket) or res.isspace():
+            res = u''
```

Comment:
Hmm… is there a cleaner way we can check for an empty disambiguation string? Perhaps, if this appears earlier, we can just use `if disam_value:` for the check.